language-plutus-core/src/Language/PlutusCore/Evaluation/Machine/ExBudgeting.hs

```diff
@@ -121,10 +122,11 @@ data ExBudgetMode =
       Counting -- ^ For precalculation
     | Restricting ExRestrictingBudget -- ^ For execution, to avoid overruns
 
--- This works nicely because @m@ contains @uni@ as parameter.
-class SpendBudget m uni | m -> uni where
+-- This works nicely because @m@ contains @term@.
+class SpendBudget m term | m -> term where
     builtinCostParams :: m CostModel
-    spendBudget :: ExBudgetCategory -> WithMemory Term uni -> ExBudget -> m ()
+    feedBudgeter :: (ExMemory -> r) -> term -> m r
```

Comment:
These methods should have some haddock. What does this do? I mean, I can guess from the types, but still.